packages/api-derive/src/balances/all.ts

```diff
@@ -114,9 +114,13 @@ function queryOld (api: ApiInterfaceRx, accountId: AccountId): Observable<Result
 
 // current (balances, vesting)
 function queryCurrent (api: ApiInterfaceRx, accountId: AccountId, balanceInstances: string[] = ['balances']): Observable<ResultBalance> {
-  const lockCalls = balanceInstances.map((m) => api.query[m].locks);
+  const lockCalls = balanceInstances.map(
+    (m): ApiInterfaceRx['query']['balances']['locks'] | undefined =>
+      (api.derive[m as 'balances'] as unknown as ApiInterfaceRx['query']['balances'])?.locks ?? api.query[m as 'balances']?.locks
+  );
+
   const lockEmpty = lockCalls.map((c) => !c);
-  const lockQueries = lockCalls.filter((c) => c).map((c): QueryableStorageMultiArg<'rxjs'> => [c, accountId]);
+  const lockQueries = lockCalls.filter((c) => c).map((c): QueryableStorageMultiArg<'rxjs'> => [c!, accountId]);
```

Comment:
Will need to think about this PR :) 

However comment here - really don't want `<var>!` anywhere in the codebase. At some point they are going to fail. Here there is no actual filter that says "only valid", so it is an assumption.

Reply:
There is a filter "(c) => c" for truthy values, so I think "!" operator cannot do much harm

Reply:
Really don't want them in the codebase at all. 

There are some that have been introduced that we need to specifically work around and that I have not managed to get out (see https://github.com/polkadot-js/api/blob/master/.eslintrc.js#L28, although I believe I cleared at leats some of them recently, need to re-check) and I really don't want to add more.

Reply:
using type predicate as a filter value seems to have done the trick